docs/source/dialogs.md

```diff
@@ -65,3 +69,22 @@ def process(self):
         ips = [entry[4][0] for entry in resolved if not '%' in entry[4][0]]
         self.produce(ResolvedHostname(name=hostname, ips=ips))
 ```
+
+### Explaining the dialogs processing mechanism during the upgrade
+
+The upgrade itself, from the operator's point of view, consists of 3 distinct stages: preupgrade, remediate and upgrade.
+
+Leapp preupgrade stage should be treated as "non-invasive system upgradeability analysis", when the upgrade workflow
+stops right after preliminary system facts collection phases and a preupgrade report containing all the information
+about potential issues is generated. If an actor containing dialog is discover during this stage, a specific
+message is added to the preupgrade report file saying that for the successful upgrade the operator should record
+their decision in the answerfile.
+
+As far as the dialogs are concerned, leapp remediate stage is intended specifically for answerfile management. The operator has the option to manually edit
```

Comment:
```suggestion
As far as the dialogs are concerned, remediation stage is intended specifically for answerfile management. The operator has the option to manually edit
```

Reply:
I meant for preupgrade\upgrade to be verbs in this context, so remediate makes more sense imho